sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -98,6 +98,19 @@ object Cast {
     case _ => false
   }
 
+  def typeCheckFailureMessage(from: DataType, to: DataType): String = (from, to) match {
+    case (_: NumericType, TimestampType) =>
+      // scalastyle:off line.size.limit
+      s"""
+         | cannot cast ${from.catalogString} to ${to.catalogString},
+         | you can enable the casting by setting ${SQLConf.LEGACY_ALLOW_CAST_NUMERIC_TO_TIMESTAMP.key}
```

Comment:
shall we remove `LEGACY_ALLOW_CAST_NUMERIC_TO_TIMESTAMP`? It's true by default and I don't see why it's needed since we have ansi mode.

Reply:
+1, we can have another PR for it

Reply:
Can we do it first? then we don't need to change this part again.

Reply:
Sure, I have just created https://github.com/apache/spark/pull/30493